<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>special_bunch_panel</class>
 <widget class="QWidget" name="special_bunch_panel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>460</width>
    <height>126</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>100</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>190</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QPushButton" name="go_to_last_bunch_in_br_pushbutton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Set the bunch number to the last bunch in the selected beam region</string>
     </property>
     <property name="text">
      <string>Last in Beam Region</string>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLabel" name="beam_region_label">
     <property name="text">
      <string>Beam Region</string>
     </property>
    </widget>
   </item>
   <item row="0" column="2">
    <widget class="QSpinBox" name="beamregion_spinbox">
     <property name="toolTip">
      <string>The beam region in which to select a diagnostic bunch</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="0" column="3">
    <widget class="QCheckBox" name="use_fast_kickers_checkbox">
     <property name="toolTip">
      <string>Whether to fire the fast kickers for the diagnostic bunch</string>
     </property>
     <property name="text">
      <string>Fast Kickers</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QPushButton" name="go_to_last_laserpulse_pushbutton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Append a bunch to the last beam region in the machine</string>
     </property>
     <property name="text">
      <string>Append Diag. Bunch</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QLabel" name="bunch_label">
     <property name="text">
      <string>Bunch</string>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QSpinBox" name="bunch_spinbox">
     <property name="toolTip">
      <string>The bunch number within the selected beam region to optionally fire the kickers or TDS for.</string>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>4000</number>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QCheckBox" name="use_tds_checkbox">
     <property name="toolTip">
      <string>Whether to fire the TDS for the diagnostic bunch</string>
     </property>
     <property name="text">
      <string>TDS</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QPushButton" name="start_button">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start diagnostic bunch optionally with fast kickers and the TDS&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { color: green; }</string>
       </property>
       <property name="text">
        <string>Start</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRepeat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="stop_button">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Start diagnostic bunch optionally with fast kickers and the TDS&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { color: red; }</string>
       </property>
       <property name="text">
        <string>Stop</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRepeat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="1">
    <widget class="QLabel" name="npulses_label">
     <property name="text">
      <string>Pulses</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QSpinBox" name="npulses_spinbox">
       <property name="toolTip">
        <string>The number of times to fire the special diagnostic bunch before stopping</string>
       </property>
       <property name="maximum">
        <number>10000000</number>
       </property>
       <property name="value">
        <number>10000</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="3">
    <widget class="QCheckBox" name="ibfb_checkbox">
     <property name="toolTip">
      <string>IBFB Adaptive FF State (Should typically be off when doing measurements)</string>
     </property>
     <property name="text">
      <string>IBFB AFF</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
